ttg/ttg/util/env.cpp

```diff
@@ -0,0 +1,33 @@
+//
+// Created by Eduard Valeyev on 11/6/21.
+//
+
+#include "ttg/util/env.h"
+
+#include <thread>
+
+#include <cstdlib>
+
+namespace ttg {
+  namespace detail {
+
+    int num_threads() {
+      std::size_t result = 0;
```

Comment:
We may want to store the result in a static variable to avoid reparsing every time this function is called (maybe not done right now, but might be in a future use-case)

Reply:
currently `initialize` is only called once so this is not an issue ... plus the user may want to change the environment variable value programmatically, so I think we must query it every time.